Added a thermal hydraulics applications page (#26860)
idaholab/moose: modules/doc/content/applications/thermal_hydraulics.md

## MOOSE-Based Applications for Thermal Hydraulics Modeling

Here we note a selection of MOOSE-based thermal hydraulics applications, which
are developed as part of the
[Nuclear Energy Advanced Modeling and Simulation (NEAMS) program](https://neams.inl.gov/).
Some of these applications are open-source, whereas some are export-controlled
and distributed via the [Nuclear Computational Resource Center (NCRC)](https://inl.gov/ncrc/);
see [help/inl/applications.md] for more information.

| Application Name                                               | Distribution | Based On                  | Added Features                                                                                                                                         |
| :------------------------------------------------------------- | :----------- | :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------- |

> | [Cardinal](https://cardinal.cels.anl.gov/) | Open-source | [NekRS](https://github.com/Nek5000/nekRS) CFD | CPU and GPU capabilities for RANS, LES, and DNS. Additional features include Lagrangian particle transport, an ALE mesh solver, overset meshes, and more. |

Reviewer: ```suggestion
| [Cardinal](https://cardinal.cels.anl.gov/) | [Open-source](https://github.com/neams-th-coe/cardinal) | [NekRS](https://github.com/Nek5000/nekRS) CFD | CPU and GPU capabilities for RANS, LES, and DNS. Additional features include Lagrangian particle transport, an ALE mesh solver, overset meshes, and more. |
```

# Thermal Hydraulics Applications

> MOOSE provides versatile, general-purpose thermal-hydraulics applications. These applications solve for mass, momentum, and energy conservation in multicomponent, multiphase flows using incompressible, weakly-compressible, or fully compressible formulations for steady-state or transients in lumped parameters and/or multidimensional (1, 2, or full 3D) geometries.

Reviewer 2: ```suggestion
MOOSE provides versatile, general-purpose thermal-hydraulics applications. These applications solve for mass, momentum, energy, and advected species conservation in multicomponent, multiphase flows using incompressible, weakly-compressible, or fully compressible formulations for steady-state or transients in lumped parameters and/or multidimensional (1, 2, or full 3D) geometries.
```

Reviewer 3: MOOSE itself is general purpose, the provided thermal-hydraulics applications are each quite specific. I propose the following:

MOOSE provides multiple thermal-hydraulic applications for varied purposes. Cumulatively, these applications solve for mass, momentum, energy, and advected species conservation in multicomponent, multiphase flows using incompressible, weakly-compressible, or fully compressible formulations for steady-state or transients in lumped parameters and/or multidimensional (1, 2, or full 3D) geometries.

| Module                                                                  | Scale                                  | Flow-Formulation                                                                                                                                             | Dimension                                                          | Typical Element Count | Typical Runtime | Typical Simulations                                                                                                                                                                                    |
| :---------------------------------------------------------------------- | :------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------------------------------- | :-------------------- | :-------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | [Navier-Stokes Modue](navier_stokes/index.md)                           | Coarse-Mesh CFD \\ \\ RANS simulations | Incompressible, Weakly-Compressible, or Fully-Compressible\\ \\ Single- or Multi-Phase \\ \\ Single- or Multi-Component Flow                                 | Typically, 2D, 2D axisymmetric, or 3D \\ \\ Can also be used in 1D | 10,000                | 1 minute        | Flow through nuclear reactor core or plena \\ \\ 3D multi-phase flow in pipes \\ \\ Natural convection flow in open cavities                                                                           |

Reviewer 4: ```suggestion
| [Navier-Stokes Module](navier_stokes/index.md)                           | Coarse-Mesh CFD \\ \\ RANS simulations | Incompressible, Weakly-Compressible, or Fully-Compressible\\ \\ Single- or Multi-Phase \\ \\ Single- or Multi-Component Flow                                 | Typically, 2D, 2D axisymmetric, or 3D \\ \\ Can also be used in 1D | 10,000                | 1 minute        | Flow through nuclear reactor core or plena \\ \\ 3D multi-phase flow in pipes \\ \\ Natural convection flow in open cavities                                                                           |
```

Reviewer 2: the run time and elements could be split ?
I dont like saying we do 1 minute runs with 10k elements
maybe we can have
10 k.                   1 minute
>100k                Setup-dependent

> | [Subchannel Module](https://subchannel-dev.hpc.inl.gov/site/index.html) | Subchannel Scale                       | Incompressible or Weakly-Compressible \\ \\ Single-Phase \\ \\ Single- or Multi-Component Flow                                                               | Typically, 3D \\ \\ Can be used in 1D and 2D                       | 100,000               | 10 seconds      | Flow development through nuclear reactor fuel assembly \\ \\ Thermal-hydraulics analysis of nuclear reactor assembly blockage \\ \\ Natural convection cooling in nuclear reactors low-flow assemblies |

Reviewer 2: this does not exist yet

add : To be released 2024

Author: I couldn't remember what we decided on this, so I didn't do anything yet. Is subchannel in the process already? I just don't want to forget to remove that comment when it does get added.

> | [Subchannel Module](https://subchannel-dev.hpc.inl.gov/site/index.html) | Subchannel Scale                       | Incompressible or Weakly-Compressible \\ \\ Single-Phase \\ \\ Single- or Multi-Component Flow                                                               | Typically, 3D \\ \\ Can be used in 1D and 2D                       | 100,000               | 10 seconds      | Flow development through nuclear reactor fuel assembly \\ \\ Thermal-hydraulics analysis of nuclear reactor assembly blockage \\ \\ Natural convection cooling in nuclear reactors low-flow assemblies |

Reviewer 3: Can be used in 1D and 2D makes no sense for subchannel. I would suggest calling it 2.5D or pseudo-3D or 3D.

Reviewer 5: As we discussed in the INL TH meeting in 03/06/24, we set for including only 3D

Author: I had written "1D with lateral coupling" - please change to "3D" if you prefer that

Author: Changed to 3D

| [Porous Flow Module](modules/porous_flow/index.md)                      | Coarse-Mesh CFD                        | Incompressible, Weakly-Compressible, or Fully-Compressible Porous Flow (no inertial term) \\ \\ Single- or Multi-Phase \\ \\ Single- or Multi-Component Flow | Typically, 2D, 2D axisymmetric, or 3D \\ \\ Can also be used in 1D | 10,000                | 1 minute        | Flow through fractured porous media \\ \\ Underground coal mining \\ \\ CO storage in saline aquifers                                                                                                  |
| [Scalar Kernels System](syntax/ScalarKernels/index.md)                  | 0D ODE                                 | Incompressible                                                                                                                                               | 0D                                                                 | 10                    | 1 second        | Bernoulli-like formulations for pressure drop                                                                                                                                                          |

> ## Open-Source Applications for Advanced Engineering Modeling
> 

Reviewer 6: ```suggestion
## Open-Source MOOSE-based Wrappers for External Fluid Dynamics Applications

```

Reviewer 6: How about this change? I agree that Cardinal is a very useful interface between Nek and everything MOOSE-based. But Nek is not MOOSE-based, and if I understand @tanoret 's proposal we are trying to advertise the tools we are working on a website we have more control over (compared to the NEAMS website). Adding advanced engineering modeling might make some visitors think that the tools mentioned before are not advanced and not for engineering.

Author: I combined all into one table now, so this suggestion probably doesn't apply anymore.

## NCRC Applications for Advanced Engineering Modeling

Via the [Nuclear Computational Resource Center (NCRC)](https://inl.gov/ncrc/), several additional applications are supported for enhanced engineering modeling. See [help/inl/applications.md] for more information.

| Application Name                                               | Based On                  | Added Features                                                                                                                                         |
| :------------------------------------------------------------- | :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------- |
| [Pronghorn](https://pronghorn-dev.hpc.inl.gov/site/index.html) | Navier-Stokes Module      | Export controlled correlations for pressure drop, heat exchange, and mass transfer in advanced nuclear reactors.                                       |

> | SAM                                                            | MOOSE Framework           | Additional lumped parameters components for realistic plant modeling and proprietary correlations for pressure drop, heat exchange, and mass transfer. |

Reviewer 7: @joshuahansel Thanks for putting this page together! Can you include this link for SAM for the time being? We'll be working on updating the webpage soon, but at least it has @rui-hu's contact info. @lingzou FYI

```suggestion
| [SAM](https://www.anl.gov/nse/system-analysis-module)                                                            | MOOSE Framework           | Additional physics and component models for realistic plant modeling and reactor safety analysis.  |
```

Reviewer: thanks for the link @travismui !

Author: Thanks! That's good that SAM has a public-facing page - we need that for some of the other apps.

> | [Sockeye](https://sockeye-dev.hpc.inl.gov/site/)               | Thermal-Hydraulics Module | Adapted correlations and specific 1D and 2D models for high-temperature heat pipes.                                                                    |

Reviewer 2: add RELAP7

### Molten Salt Reactors

!media thermal_hydraulics/misc/example_msr.png
       style=width:90%;display:block;margin-left:auto;margin-right:auto;
       caption=RANS simulation of conjugated heat transfer in a pool-type molten salt reactor concept using the MOOSE Navier-Stokes module.
       id=mcre

!media thermal_hydraulics/misc/example_msre.png
       style=width:90%;display:block;margin-left:auto;margin-right:auto;
       caption=Power density (left), fuel temperature (center), and void fraction distribution (right) during the steady-state operation of the Molten Salt Reactor Experiment using the MOOSE Navier-Stokes module RANS simulation.
       id=msre

!col-end!

!col! small=4 medium=4 large=4

### High Temperature Reactors

!media thermal_hydraulics/misc/example_httf.png
       style=width:90%;display:block;margin-left:auto;margin-right:auto;
       caption=Steady-state operation of Oregon State University's High Temperature Test Facility (HTTF) using the MOOSE Navier-Stokes module coarse-mesh CFD capability.
       id=httf

!media thermal_hydraulics/misc/example_httr.png
       style=width:90%;display:block;margin-left:auto;margin-right:auto;

>        caption=Temperature in fuel blocks of the High-Temperature Engineering Test Reactor (HTTR) during steady-state operation using the MOOSE Navier-Stokes module coarse-mesh CFD capability.

Reviewer 4: ```suggestion
       caption=Temperature in fuel blocks of the High-Temperature Engineering Test Reactor (HTTR) during steady-state operation using the MOOSE Thermal Hydraulics Module (THM).
```

# Thermal Hydraulics Applications

There are several MOOSE-based applications modeling thermal hydraulics, which
are developed as part of the
[Nuclear Energy Advanced Modeling and Simulation (NEAMS) program](https://neams.inl.gov/).
These applications provide capabilities at various levels of fidelity vs.
computational efficiency for a multi-layered simulation strategy. The following
table summarizes these applications, in order of decreasing fidelity.

> | Application | Description | Link(s) |

Reviewer 2: I would either split this table in open & not open source

also NS module should be featured?

OR add a column
